spec.md

```diff
@@ -570,13 +559,297 @@ schema being used.
+          "update": {
+            "parameters": {
+              "$schema": "http://json-schema.org/draft-04/schema#",
+              "type": "object",
+              "propeties": {
+                "some-other-account": {
+                  "description": "Other fake account needed for update.",
+                  "type": "string"
+                },
+                "billing": {
+                  "$ref": "osb_v2:///fakeservice/v1#BillingAccountParameters"
+                }
+              }
+            }
+          }
+        },
+        "service_binding": {
+          "create": {
+            "parameters": {
+              "$ref": "osb_v2:///fakeservice/v1/bindings#AccountParameters"
+            }
+          }
+        }
       }
     }]
   }]
 }
 ```
 
 
+## JSON Schemas
+
+The JSON Schemas endpoint is used to define reusable definitions of JSON Schema
+Objects referenced throughout the [Catalog](#catalog-management). If a Platform
+intends to support JSON Schemas, it MUST examine the response from the catalog
+to determine if the Service Broker is using `$ref` JSON Schemas. The Service
+Broker can signal to the Platform that JSON Schema components are supported by
+using the `osb_v2:` scheme. Components are a way to allow Platforms to only
+fetch sub-schemas.
+
+The root JSON Schema document can be found by getting the JSON Schemas
+endpoint without a `:component_path`.
+
+If the `osb_v2:` scheme is used by the Service Broker, then the Platform MAY
+fetch the component JSON Schema directly using the following convention:
+
+ - Given: `osb_v2:///{:component_path}`, fetch
+   `/v2/catalog/schemas/:component_path` and then the Platform is responsible
+   to continue fetching any unknown `:component_path` URIs. Care ought to be
+   taken by the platform to not fetch duplicate definitions.
+
+A Platform MAY ignore the `osb_v2:` scheme and just fetch `/v2/catalog/schemas`
+which will contain all sub-Schema documents. This reduces the complexity of the
+Platform but increases the potential payload size of the JSON Schemas.
+
+A Platform SHOULD fetch the root Schema document for any other `$ref` URIs
+found that do not match the `osb_v2:` scheme. The Service Broker MUST provide
+the JSON Schema Object for the referenced URI in a subschema returned.
+
+Note: If the Service Broker provides a mix of `osb_v2:` in addition to other
+schemes that result in fetching both `/v2/catalog/schemas/:component_path` and
+`/v2/catalog/schemas`, the Platform will have colliding JSON Schema `$ids`. In
+this case the Platform SHOULD ignore the component paths and use the root
+schema. Service Brokers SHOULD avoid mixing `osb_v2:` and other schemes.
+
+When the osb_v2: scheme is used, the data referenced by a component path MUST
+NOT change and can only be deleted when there are no longer any references to
+it in the Service Broker's schema. Additionally, once deleted, that component
+path MUST NOT ever be used again by this Service Broker, unless the schema
```

Comment:
which requirement leads to this strong constraint of infinite memory in the broker to never reuse schema component paths? Is there need for unlimited caching by the platforms?

Reply:
The path can be fetched again, this requirement is for serving an object that does not match the previously served object from the same url. This is the spec way to say use versions without saying directly that the objects need to be versioned. for example, a one-off broker does not want to worry about versions, but their resources are essentially versioned from the fact that they are static. This gives the rule that they need to not serve a new object from the same url.